src/view/screens/CustomFeed.tsx

```diff
@@ -34,17 +34,41 @@ import {EmptyState} from 'view/com/util/EmptyState'
 import {useAnalytics} from 'lib/analytics/analytics'
 import {NativeDropdown, DropdownItem} from 'view/com/util/forms/NativeDropdown'
 import {makeProfileLink} from 'lib/routes/links'
+import {resolveName} from 'lib/api'
 
 type Props = NativeStackScreenProps<CommonNavigatorParams, 'CustomFeed'>
-export const CustomFeedScreen = withAuthRequired(
-  observer(({route}: Props) => {
+
+export const CustomFeedScreen = observer((props: Props) => {
+  const store = useStores()
+  const {name: handleOrDid} = props.route.params
+  const [feedOwnerDid, setFeedOwnerDid] = React.useState<string | undefined>()
+
+  React.useEffect(() => {
+    /*
+     * We must resolve the DID of the feed owner before we can fetch the feed.
+     */
+    async function fetchDid() {
+      const did = await resolveName(store, handleOrDid)
+      setFeedOwnerDid(did)
+    }
+
+    fetchDid()
+  }, [store, handleOrDid, setFeedOwnerDid])
+
+  return feedOwnerDid ? (
+    <CustomFeedScreenInner {...props} feedOwnerDid={feedOwnerDid} />
+  ) : null
```

Comment:
Probably should put a loading spinner or something here, it's entirely possible the network call could hang and this would leave it on a blank screen